lib/core_extensions.rb

```diff
@@ -39,8 +39,14 @@ def initialize(*attribute)
 
     def before_destroy(record)
       klasses.each do |klass|
-        record.send(klass.to_sym).each do |what|
-          what = what.to_label unless what.is_a? String
+        record.association(klass.to_sym).association_scope.each do |what|
```

Comment:
This may cause a problem in Rails 4.  I'd have to double check, but I'm pretty sure that associations now inherit the default scope so you'd have to instead explicitly call unscoped.